.pre-commit-config.yaml

```diff
@@ -1,25 +1,48 @@
+default_stages: [pre-commit, pre-push]
+default_install_hook_types: [pre-commit, pre-push]
 repos:
 - repo: https://github.com/gitleaks/gitleaks
   rev: v8.21.1
   hooks:
   - id: gitleaks
+    stages: [pre-commit]
 - repo: https://github.com/koalaman/shellcheck-precommit
   rev: v0.10.0
   hooks:
   - id: shellcheck
     args: ["--external-sources"]
+    stages: [pre-commit]
 - repo: https://github.com/pre-commit/pre-commit-hooks
   rev: v5.0.0
   hooks:
   - id: end-of-file-fixer
+    stages: [pre-commit]
   - id: trailing-whitespace
+    stages: [pre-commit]
   - id: detect-private-key
+    stages: [pre-commit]
 - repo: https://github.com/pylint-dev/pylint
   rev: v3.3.1
   hooks:
   - id: pylint
     exclude: ^hack/boilerplate/boilerplate.py$
+    stages: [pre-commit]
 - repo: https://github.com/jumanjihouse/pre-commit-hooks
   rev: 3.0.0
   hooks:
-    - id: forbid-binary
+  - id: forbid-binary
+    stages: [pre-commit]
+- repo: local
+  hooks:
+  - id: make-verify
+    name: Run make verify
+    entry: make verify
+    language: system
+    stages: [pre-push]
+- repo: local
+  hooks:
+  - id: make-test
+    name: Run make test
+    entry: make test
+    language: system
+    stages: [pre-push]
```

Comment:
Same thought here. `make test` atm seems to be running `generate` as well. Ref: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/127514395eb0c6c7ec20eff9ff29fdba1a941793/Makefile#L701

In my experience, `make generate` takes a lot of time to finish. 
- I wonder if we can drop `generate` as a dependency? **Or** 
- Maybe come up with a new make target that runs UT exclusively ? i.e. runs `make go-test-race`. So essentially we run `make go-test-race` in this hook. Although I have not tested the side affects of running `make go-test-race` by itself.

However, this comment can be a followed up another issue-PR.